File: app/src/main/java/com/constraint/vagabond/data/local/RecAreaDatabase.java
package com.constraint.vagabond.data.local;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import com.constraint.vagabond.data.entities.DataConverter;
import com.constraint.vagabond.data.entities.RecreationalArea;

@Database(
    entities = {RecreationalArea.class},
    version = 1,
    exportSchema = false)
@TypeConverters(DataConverter.class)
public abstract class RecAreaDatabase extends RoomDatabase {

  private static final Object sLock = new Object();
  private static RecAreaDatabase INSTANCE;

  public static RecAreaDatabase getInstance(final Context context) {
    synchronized (sLock) {
      if (INSTANCE == null) {
        INSTANCE =
            Room.databaseBuilder(context.getApplicationContext(), RecAreaDatabase.class, "Areas.db")
                .build();
      }
    }
    return INSTANCE;
  }

  public abstract RecAreaDao recAreaDao();

  public void destroyDB() {
    INSTANCE = null;
  }
}

File: app/src/main/java/com/sierra/vagabond/data/local/RecAreaDatabase.kt
package com.sierra.vagabond.data.local

import android.app.Application
import android.content.Context

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

import com.sierra.vagabond.data.entities.RecreationalArea
import com.sierra.vagabond.utils.DB_NAME

@Database(entities = [RecreationalArea::class], version = 2, exportSchema = false)
@TypeConverters(DataConverter::class)
abstract class RecAreaDatabase : RoomDatabase() {

    abstract fun recAreaDao(): RecAreaDao

    fun destroyDB() {
        INSTANCE = null
    }

    companion object {

        @Volatile
        private var INSTANCE: RecAreaDatabase? = null

        fun getInstance(application: Application): RecAreaDatabase =
                INSTANCE ?: synchronized(this) {
                    INSTANCE ?: buildDatabase(application).also { INSTANCE = it }
                }

        private fun buildDatabase(context: Context) =
                Room.databaseBuilder(context.applicationContext,
                        RecAreaDatabase::class.java, DB_NAME)
                        .fallbackToDestructiveMigration()
                        .build()
    }
}

